apps/transformers/lib/transformations/concatenation.ex

```diff
@@ -17,21 +23,20 @@ defmodule Transformers.Concatenation do
   end
 
   def validate(parameters) do
-    with {:ok, source_fields} <- FieldFetcher.fetch_parameter(parameters, "sourceFields"),
-         {:ok, separator} <- FieldFetcher.fetch_parameter(parameters, "separator"),
-         {:ok, target_field} <- FieldFetcher.fetch_parameter(parameters, "targetField") do
-      {:ok, [source_fields, separator, target_field]}
-    else
-      {:error, reason} -> {:error, reason}
-    end
+    %ValidationStatus{}
+    |> NotBlank.check(parameters, @source_fields)
+    |> NotBlank.check(parameters, @target_field)
+    |> IsPresent.check(parameters, @separator)
+    |> ValidationStatus.ordered_values_or_errors([@source_fields, @separator, @target_field])
   end
 
   def fetch_values(payload, field_names) when is_list(field_names) do
     find_values_or_errors(payload, field_names)
     |> all_values_if_present_else_error()
   end
 
-  def fetch_values(_, _), do: {:error, "Expected list but received single value: sourceFields"}
+  def fetch_values(_, _),
+    do: {:error, "Expected list but received single value: #{@source_fields}"}
```

Comment:
Is this helpful for debugging? Could we instead show the value that didn't match?

Reply:
The thing I changed here was using the new module attribute/string constant consistently throughout the file, and then mix format changed the indentation. I'd prefer to leave it for now, because I think giving the payload validations a refactoring when we have time would be beneficial.